test/pool.test.ts

```diff
@@ -5,13 +5,19 @@ import {
 import {Pool} from "../contract-bindings/ethers/Pool";
 import {ReceiverWeightsTest} from "../contract-bindings/ethers/ReceiverWeightsTest";
 import buidler from "@nomiclabs/buidler";
-import {Signer} from "ethers";
+import {Signer, BigNumber, BigNumberish, ethers} from "ethers";
 import {assert, expect} from "chai";
 
 async function addr(idx: number): Promise<string> {
   return await (await buidler.ethers.getSigners())[idx].getAddress();
 }
 
+function numberToAddress(num: number): string {
```

Comment:
We could call this `generateAddress()` and randomly generate an address instead of using the `num` parameter. We don’t rely on this to be deterministic and if we need to reuse it we can always save the address in a local variable

Reply:
We do, we generate explicit addresses for `0` and `1`. Random addresses are indeed a little cleaner and easier to use, added `randomAddress` and switched to it wherever possible.